### DISCUSSION TITLE
RFC: Astro.fetchContent('./posts/**/*.astro')

### DISCUSSION THREAD

**Discussion by MattWilcox** [withastro/roadmap#7] [Proposal]
Moving this from the [old RFC](https://github.com/withastro/astro/issues/1497) to the new Discussions area.
Still would love this, as I see no reason (from an author perspective) that we can't use fetchContent with *.astro files, meaning I have to use MarkDown anywhere I'd want to do fetchContent - which isn't always what I want given the different capabilities of markdown vs astro...

---

### Background & Motivation
I want to support the workflow used in https://github.com/snowpackjs/astro/tree/main/www

Blog posts are written in www/src/pages/blog/[*.astro], and uses <Markdown>.

https://github.com/snowpackjs/astro/blob/main/www/src/pages/blog/index.astro lists all of the posts in a list, which is hardcoded today!

I want to support let posts = Astro.fetchContent('./pages/blog/**.astro'); as a use-case,

This will tie in with
💡 RFC: Allow exports from astro components #1142

### Possible solutions
- just use import.meta.globEager
- add astro files to the allowed files in Astro.fetchContent

**Comment by florian-lefebvre** [comment]
This was implemented as `Astro.glob`, then deprecated in v5 in favor of `import.meta.glob`
